### ISSUE TITLE
Can't start Golem: argument must be a string, a bytes-like object or a number, not 'NoneType'

### ISSUE THREAD

**Issue by FotG** [golemfactory/clay#2923]
Attempting to start Golem fails with the client indicating a problem with ports. I am on v0.1.12

The output of  golem.error.log:
```
2018-06-23 01:46:44 WARNING  golem_sci.implementation            b'Unable to use `Faucet` contract'
2018-06-23 01:46:44 WARNING  golem_sci.implementation            b'Unable to use `GNTDeposit` contract'
2018-06-23 01:46:44 WARNING  golem_sci.implementation            b'Unable to use `GNTPaymentChannels` contract'
2018-06-23 01:46:45 ERROR    golem.node                          b'Stopping because of 'setup client' error: <twisted.python.failure.Failure builtins.TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'>'
2018-06-23 01:47:06 WARNING  twisted                             session closed with reason wamp.close.transport_lost [()]
```
The same message is also displayed in golem.log:
```
2018-06-23 01:45:20 INFO     talkback                            b'Disabling talkback 'sentry' service'
2018-06-23 01:45:20 INFO     talkback                            b'Disabling talkback 'sentry-metrics' service'
2018-06-23 01:45:20 INFO     golemapp                            b'GOLEM Version: 0.16.0'
2018-06-23 01:45:20 INFO     golemapp                            b'Protocol Version: 27'
2018-06-23 01:45:20 INFO     golemapp                            b'golem_messages Version: 2.9.4'
2018-06-23 01:45:20 INFO     golemapp                            b'system: Windows, release: 10, version: 10.0.17134, machine: AMD64'
2018-06-23 01:45:35 INFO     golemapp                            b'cpu: GenuineIntel Intel(R) Core(TM) i7-5960X CPU @ 3.00GHz, 16 cores'
2018-06-23 01:45:35 INFO     golemapp                            b'memory: 15.9 GiB, swap: 18.3 GiB'
2018-06-23 01:45:35 INFO     golemapp                            b'Ethereum chain: mainnet'
2018-06-23 01:45:35 INFO     golemapp                            b'Concent disabled'
2018-06-23 01:45:46 INFO     golem.node                          b'Waiting for password to unlock the account. Run `golemcli account unlock` and enter your password.'
2018-06-23 01:45:52 INFO     golem.docker.manager                b'DockerMachine: starting golem'
2018-06-23 01:45:56 INFO     golem.node                          b'Got password'
2018-06-23 01:46:42 INFO     golem.docker.manager                b'DockerMachine: env updated'
2018-06-23 01:46:43 INFO     talkback                            b'Disabling talkback 'sentry' service'
2018-06-23 01:46:43 INFO     talkback                            b'Disabling talkback 'sentry-metrics' service'
2018-06-23 01:46:43 INFO     golem.client                        b'Client "FotG", datadir: C:\Users\[removed]\AppData\Local\golem\golem\default\mainnet'
2018-06-23 01:46:43 INFO     golem.core.hardware                 b'updating config: name: custom, num_cores: 15, max_memory_size: 11.9 GiB, max_resource_size: 1.0 GiB'
2018-06-23 01:46:43 INFO     golem.pay                           b'Node Ethereum address: [removed]
2018-06-23 01:46:43 INFO     golem.ethereum                      b'GETH: connecting to remote RPC interface at https://2.geth.golem.network:55555'
2018-06-23 01:46:44 INFO     golem.ethereum                      b'Connected to node in 0.6798276901245117s'
2018-06-23 01:46:44 WARNING  golem_sci.implementation            b'Unable to use `Faucet` contract'
2018-06-23 01:46:44 WARNING  golem_sci.implementation            b'Unable to use `GNTDeposit` contract'
2018-06-23 01:46:44 WARNING  golem_sci.implementation            b'Unable to use `GNTPaymentChannels` contract'
2018-06-23 01:46:45 ERROR    golem.node                          b'Stopping because of 'setup client' error: <twisted.python.failure.Failure builtins.TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'>'
2018-06-23 01:47:06 WARNING  twisted                             session closed with reason wamp.close.transport_lost [()]
```

gui.log:
```
2018-06-23T07:45:48.671Z - warn: Fetch golem.status failed! wamp.error.no_such_procedure {} no callee registered for procedure <golem.status> location=SAGA > HANDLER
2018-06-23T07:45:51.433Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:45:51.912Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:45:52.474Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:45:52.478Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:45:56.434Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:46:39.752Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:46:39.755Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:46:42.769Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:46:45.493Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:46:45.497Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:46:45.500Z - warn: TypeError {} location=SAGA > GOLEM
2018-06-23T07:47:06.978Z - info: Golem exited with code 0 location=MAIN_PROCESS > GOLEM_HANDLER
```

**Comment by FotG** [comment]
I did get Golem up and running, but I had to uninstall and reinstall. First time I installed I didn't have send statistics enabled. The second time I left it at its default value of enabled. I don't know if that is any help.

**Comment by Wiezzel** [comment]
@FotG What operating system are you running? How do you start Golem (clicking an icon or by command-line)?

**Comment by FotG** [comment]
I am on Windows 10 x64 1803, and I launched it from the Icon for the mainnet.

**Comment by Wiezzel** [comment]
@FotG Thank you for feedback. I will investigate the issue.

**Comment by Wiezzel** [comment]
I could not reproduce this error on my machine (Windows 10) using 0.16.0 binaries.
From the provided logs I infer that the error occurred in this code fragment:
https://github.com/golemfactory/golem/blob/83bfaf3d15c950498b6f5985e615d838200b736c/golem/transactions/ethereum/ethereumtransactionsystem.py#L65-L67
I tried to analyze all calls to `int()` from these lines but there were around 100 (!) of them. Further debugging is impossible without the full stacktrace which we unfortunately lack. In #2933 I introduced an extended logging mechanism that will print the stacktrace if any unhandled error occurs during node initialization.
